core/session.ts

```diff
@@ -445,6 +445,8 @@ export class Session {
       );
     }
 
+    this.removeNonUniqueActionsFromCompiledGraph(standardActions, compiledGraph)
```

Comment:
I think this should come immediately after the call to `alterActionName` - ideally we would stop mutating the graph before running any of the `check` methods.

Reply:
[checkActionNameUniqueness](https://github.com/dataform-co/dataform/blob/main/core/session.ts#:~:text=private-,checkActionNameUniqueness,-(actions%3A) and [checkCanonicalTargetUniqueness](https://github.com/dataform-co/dataform/blob/main/core/session.ts#:~:text=private-,checkCanonicalTargetUniqueness,-(actions%3A) detect non-unique actions and each of them include N - 1 corresponding compilation errors (N - number of occurrences of a duplicate action). So if we remove duplicate actions before these check methods starts they will not detect any duplicate actions. Should I attach those errors when removing duplicate actions from the graph and remove these check methods?

Reply:
Hmm, good point, I'd missed that! Either way is OK with me, it's probably a bit more obvious to the reader to do as you suggest, but also happy with the code as-is!